src/Text-Core/RunArray.class.st

```diff
@@ -522,7 +488,8 @@ RunArray >> includes: anObject [
 
 { #category : #'self evaluating' }
 RunArray >> isSelfEvaluating [
-	^ self class == RunArray and: [values isSelfEvaluating]
+
+	^ values isSelfEvaluating
```

Comment:
I think the test is a guard to prevent instance of subclasses to be wrongfully declared as self evaluating. I think I saw those in other classes.
Better be safe than sorry here, bugs caused here will be hard to track.

Reply:
As I said in a previous comment, this is an internal class of Text. I don't think it's good to subclass this kind of classes. You don't want to rely on the internal implementation of other stuff

Reply:
If it is planned to move this class to Collections pacakage, I agree with Jean to keep the guard